dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/decorations/BuiltIn.java

```diff
@@ -44,21 +59,43 @@ public void addDecoration(String id, String... params) {
         values.add(params[0]);
     }
 
-    public void decorate(String id, MemoryObject memObj, Type type) {
-        if (x < 0 || y < 0 || z < 0) {
-            throw new ParsingException("Illegal BuiltIn hierarchy ('%d', '%d', '%d')", x, y, z);
-        }
+    public Expression decorate(String id, Expression expr, Type type) {
         for (String decoration : mapping.getOrDefault(id, List.of())) {
             switch (decoration) {
                 case "LocalInvocationId", "GlobalInvocationId" -> {
-                    int size = TypeFactory.getInstance().getMemorySizeInBytes(type) / 3;
-                    // TODO: Use array element type instead of arch type
-                    memObj.setInitialValue(0, FACTORY.makeValue(x, TYPE_ARCH));
-                    memObj.setInitialValue(size, FACTORY.makeValue(y, TYPE_ARCH));
-                    memObj.setInitialValue(size * 2, FACTORY.makeValue(z, TYPE_ARCH));
+                    if (x < 0 || y < 0 || z < 0) {
+                        throw new ParsingException("Illegal BuiltIn hierarchy ('%d', '%d', '%d')", x, y, z);
+                    }
```

Comment:
isn't zero also a wrong value for hierarchy?

Reply:
Yeah. I'll update it